.github/workflows/documentation.yml

```
@@ -0,0 +1,41 @@
+name: Documentation
+on:
+  push:
+    branches:
+    - main
+
+jobs:
+  docs:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        fetch-depth: 0
+        lfs: true
+    - name: Install Python
+      uses: actions/setup-python@v2
+      with:
+        python-version: '3.x'    
+    - uses: ammaraskar/sphinx-action@master
+      with:
+        pre-build-command: "pip install camel-converter[pydantic] && pip install sphinx_rtd_theme && sphinx-apidoc -f -o docs meilisearch/"
```

I would add the sphinx requirements as dev requirements to the project rather than installing requirements here. Then install all requirements from the project. Installing requirements in the action is going to be error prone because any time a new requirement is added it also has to be added here, versions may not match, etc.

I have not used the sphinx action before so my syntax may be off here.

```suggestion
    - name: Install pipenv
      uses: dschep/install-pipenv-action@v1
    - name: Install dependencies
      run: pipenv install --dev
        
    - uses: ammaraskar/sphinx-action@master
      with:
        pre-build-command: "sphinx-apidoc -f -o docs meilisearch/"
```

---

I have changed this with pipenv, but I tested and the 'sphinx_rtd_theme' was not being picked up in the action. So, i kept that the same way and installed everything else with pipenv.